airflow/providers/docker/decorators/docker.py

```diff
@@ -79,10 +79,6 @@ class _DockerDecoratedOperator(DecoratedOperator, DockerOperator):
 
     template_fields: Sequence[str] = (*DockerOperator.template_fields, "op_args", "op_kwargs")
 
-    # since we won't mutate the arguments, we should just do the shallow copy
-    # there are some cases we can't deepcopy the objects (e.g protobuf).
-    shallow_copy_attrs: Sequence[str] = ("python_callable",)
```

Comment:
I double checked, and this was introduced in 2.1 with https://github.com/apache/airflow/pull/14709 so there's no version issues we have to worry about here.